include/envoy/http/header_map.h

```diff
@@ -781,5 +782,22 @@ class HeaderMatcher {
 
 using HeaderMatcherSharedPtr = std::shared_ptr<HeaderMatcher>;
 
+/**
+ * An interface to be implemented by rate limited reset header parsers.
+ */
+class ResetHeaderParser {
+public:
+  virtual ~ResetHeaderParser() = default;
+
+  /**
+   * Iterate over the headers, choose the first one that matches by name, and try to parse its
+   * value.
+   */
+  virtual absl::optional<std::chrono::milliseconds>
+  parseInterval(TimeSource& time_source, const HeaderMap& headers) const PURE;
+};
+
+using ResetHeaderParserSharedPtr = std::shared_ptr<ResetHeaderParser>;
```

Comment:
I would recommend putting this closer to the code where it is used. I don't think it needs to be in the main header map interface.

Reply:
It's used in the RetryPolicy interface, which is defined in `include/envoy/router/router.h`. Would it be better to move it into that header instead?

Reply:
Sure I think that would be better.

Reply:
Great, I moved it.